ogre/include/ignition/rendering/ogre/OgreDistortionPass.hh

```diff
@@ -0,0 +1,146 @@
+      /// \param[in] _k2 Radial distortion coefficient k2.
+      /// \param[in] _k3 Radial distortion coefficient k3.
+      /// \param[in] _p1 Tangential distortion coefficient p1.
+      /// \param[in] _p2 Tangential distortion coefficient p2.
+      /// \param[in] _width Width of the image texture in pixels.
+      /// \param[in] _f Focal length in pixels.
+      /// \return Distorted coordinate.
+      public: static ignition::math::Vector2d Distort(
+                  const ignition::math::Vector2d &_in,
+                  const ignition::math::Vector2d &_center,
+                  double _k1, double _k2, double _k3,
+                  double _p1, double _p2,
+                  unsigned int _width, double _f);
+
+      /// \brief get the distortion map value.
+      /// \param[in] _x X component of map.
+      /// \param[in] _y Y component of map.
+      /// \return the distortion map value at the specified index.
+      protected: ignition::math::Vector2d
+        DistortionMapValueClamped(const int _x, const int _y) const;
+
+      /// \brief calculate the correct scale factor to "zoom" the render,
+      /// cutting off black borders caused by distortion (only if the crop
+      /// flag has been set).
+      protected: void CalculateAndApplyDistortionScale();
+
+      /// \brief Scale applied to distorted image.
+      public: ignition::math::Vector2d distortionScale = {1.0, 1.0};
+
+      /// \brief True if the distorted image will be cropped to remove the
+      /// black pixels at the corners of the image.
+      public: bool distortionCrop = true;
+
+      /// \brief Use the legacy distortion mode. If this is set to false, the
+      /// new mode modifies how Brown's distortion equations are applied to
+      /// better reflect real distortion. The image is projected from image
+      /// plane to camera plane to apply distortion equations, then projected
+      /// back to image plane. Note that the new distortion doesn't allow
+      /// the image to be cropped.
+      public: bool legacyMode = true;
+
+      /// \brief Distortion compositor.
+      public: Ogre::CompositorInstance *distortionInstance = nullptr;
+
+      /// \brief Ogre Material that contains the distortion shader
+      public: Ogre::MaterialPtr distortionMaterial;
+
+      /// \brief Mapping of distorted to undistorted normalized pixels
+      public: std::vector<ignition::math::Vector2d> distortionMap;
+
+      /// \brief Width of distortion texture map
+      public: unsigned int distortionTexWidth;
+
+      /// \brief Height of distortion texture map
+      public: unsigned int distortionTexHeight;
+
+      /// \brief Distortion compositor listener
+      /// uses <ignition/utils/ImplPtr.hh> from ign-rendering7
+      IGN_COMMON_WARN_IGNORE__DLL_INTERFACE_MISSING
+      public: std::shared_ptr<DistortionCompositorListener>
+          distortionCompositorListener;
+      IGN_COMMON_WARN_RESUME__DLL_INTERFACE_MISSING
```

Comment:
can you add a private data object `dataPtr` and move the member variables to that class? Similar to the [dataPtr](https://github.com/ignitionrobotics/ign-rendering/blob/main/include/ignition/rendering/CameraLens.hh#L185) here. This is so that we it's easier to maintain ABI if we want to add new variables in the future.